### PR TITLE
Rewrite mappers FormTemplates to be components.

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- rewrite mappers FormTemplates to be components.
- onStateUpdate is back as a FormRenderer prop

### CODE REVIEW

File: packages/react-form-renderer/src/components/form-renderer.js [hunk 1]
```diff
@@ -55,38 +82,43 @@ const FormRenderer = ({
...
-              renderForm,
-              ...mutators,
-              ...form
-            }
-          }}
-        >
-          <FormTemplate formFields={renderForm(schema.fields)} schema={schema} />
-        </RendererContext.Provider>
-      )}
+      render={({ handleSubmit, pristine, valid, form: { reset, mutators, getState, submit, ...form }, ...state }) => {
```
Comment (rvsia) [inline packages/react-form-renderer/src/components/form-renderer.js]: Unnecessary return block

File: packages/react-form-renderer/src/components/form-renderer.js [hunk 2]
```diff
@@ -10,9 +10,35 @@ import defaultSchemaValidator from '../parsers/default-schema-validator';
 import SchemaErrorComponent from '../form-renderer/schema-error-component';
 import defaultValidatorMapper from '../validators/validator-mapper';
 
+const useComponentSpy = (onStateUpdate) => {
```
Comment (rvsia) [inline packages/react-form-renderer/src/components/form-renderer.js]: What about to move this to hooks folder?

Comment (Hyperkid123) [reply packages/react-form-renderer/src/components/form-renderer.js]: I could. If you reaaaaaally want me to

Comment (rvsia) [reply packages/react-form-renderer/src/components/form-renderer.js]: Yes, please :D

File: packages/react-form-renderer/src/tests/form-renderer/render-form.test.js [hunk 3]
```diff
@@ -1185,7 +1185,7 @@ describe('renderForm function', () => {
     });
   });
 
-  it('should use actionMapper', () => {
+  it.only('should use actionMapper', () => {
```
Comment (rvsia) [inline packages/react-form-renderer/src/tests/form-renderer/render-form.test.js]: only ?

**Pull request merged**